### ISSUE TITLE
Add more fields in Instrument class

### ISSUE THREAD

**Issue by rsmb7zhx** [nautechsystems/nautilus_trader#851]
# Feature Request

Hi,
It would be great to include following fields, in particular HeadTimeStamp to be stored in the Instrument class as it can be of common use. Also considering the fact that historical data requests are costly (in terms of pacing etc), to avoid making unncessary requests before the instrument was first available and also that will be useful for analysing the reason for not having backfill data before particular date of the instrument.

- Head TimeStamp
- Trading Hours
- Liquid Hours
- Time Zone

**Comment by rsmb7z** [comment]
I found there is `info: dict` field available in base `Instrument` class but not implemented for `Equity`, `Future`, `Option`. I think if that is added in these shall suffice to store any information depending on Provider's availability without the need to add more fields to Instrument.

**Comment by cjdsellers** [comment]
I see that they are implemented though? so the intent of those methods is to aid serialization (starting with a Python dict of 'primitives' makes things easier here) - so I think its best if these dicts provide a full representation of the object.

So we should probably just add the necessary fields to the instruments (optional where needed).

I'm guessing some of these you propose can be fetched through the IB API, which ones specifically are you interested in? and if you could provide a link to the relevant IB docs that would be very helpful! cheers

**Comment by rsmb7z** [comment]
> I see that they are implemented though? so the intent of those methods is to aid serialization (starting with a Python dict of 'primitives' makes things easier here) - so I think its best if these dicts provide a full representation of the object.

Yes the `info` is implemented and as `dict` provides flexibility to store all `contractDetails` in my case. However what I mean is it is not exposed to Equity, Future, and Option. There is init kwarg error when adding `info` to these instrument types.
Just to point out, its mutable and `Cache` may need to return copy of dict when returning instrument.



> I'm guessing some of these you propose can be fetched through the IB API, which ones specifically are you interested in? and if you could provide a link to the relevant IB docs that would be very helpful! cheers

*Trading Hours, Liquid Hours, and Time Zone* is available within ContractDetails which is loaded to create the Instrument and doesn't cost any additional requests. While exploring I found there is `MarketStatus` enum available - still going through though on how to wire that in. So these three becomes more related to `MarketStatus` however apart from generic, we have here per Instrument Market Info. I will get back with more info on this.

*Head TimeStamp*, I will say to opt this out because:
- It requires seperate request to get this info
- Each PriceType will require seperate and may have different date
- Is applicable for Backtesting when fetching start request
